### PR TITLE
Remove queryLimits.maxResults, add maximumTake

### PR DESCRIPTION
**Pull request by dcousens**
This pull request removes `context.totalResults` and `context.maxTotalResults`.

This state was not carried between `.sudo`, `.exitSudo`, or `.withSession` methods, and is not actually useful for anything except some broad/inaccurate statistics.  You should not be using these for any form of security or session control.

I recommend we remove the feature completely, if developers want to track a session, they can rely on a custom session strategy.
If developers want to track access to items, they can use the access control functions.

This pull request additionally adds `[list].graphql.maximumTake`, a list configuration option to enforce the maximum `take` value in `findMany` queries for that list.  This functionality is unambiguous and has clear behavioral boundaries.

### TODO
- [x] Implementation
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: docs/pages/docs/config/config.md [hunk 1]
```diff
@@ -386,8 +386,6 @@ Options:
 
 - `debug` (default: `process.env.NODE_ENV !== 'production'`): If `true`, stacktraces from both Apollo errors and Keystone errors will be included in the errors returned from the GraphQL API.
   These can be filtered out with `apolloConfig.formatError` if you need to process them, but do not want them returned over the GraphQL API.
-- `queryLimits` (default: `undefined`): Allows you to limit the total number of results returned from a query to your GraphQL API.
```
Comment (dcousens) [inline docs/pages/docs/config/config.md:389]: This was removed many moons ago

File: packages/core/src/lib/core/queries/resolvers.ts [hunk 2]
```diff
@@ -113,6 +113,11 @@ export async function findMany(
   info: GraphQLResolveInfo,
   extraFilter?: PrismaFilter
 ): Promise<BaseItem[]> {
+  const maximumTake = (list.types.findManyArgs.take?.defaultValue ?? Infinity) as number;
```
Comment (mitchellhamilton) [inline packages/core/src/lib/core/queries/resolvers.ts:116]:
```suggestion
  const maximumTake = (list.types.findManyArgs.take.defaultValue ?? Infinity) as number;
```

**Review by mitchellhamilton — COMMENTED** [review]
Docs changes need conditionals (at least for the additions)

File: packages/core/src/types/config/lists.ts [hunk 3]
```diff
@@ -204,12 +204,11 @@ export type ListGraphQLConfig = {
...
-  // was previously top-level listQueryName
   plural?: string;
-  // was previously top-level queryLimits
-  queryLimits?: {
-    maxResults?: number; // maximum number of items that can be returned in a query (or subquery)
-  };
+  /**
+   * The maximum value for the take parameter when querying this list
+   */
+  maximumTake?: number;
```
Comment (mitchellhamilton) [inline packages/core/src/types/config/lists.ts:211]: We use `max` in the `validation` config on fields, thoughts on `maxTake`?

File: docs/pages/docs/config/lists.md [hunk 4]
```diff
@@ -104,7 +104,7 @@ Options:
   - `initialSort` (default: `undefined`): Sets the field and direction to be used to initially sort the data in the list view.
     Option `field` is the name of the field to sort by, and `direction` is either `'ASC'` or `'DESC'` for ascending and descending sorting respectively.
     If undefined then data will be unsorted.
-  - `pageSize` (default: `50`): Sets the number of items to show per page in the list view.
+  - `pageSize` (default: lower of `50` or [`graphql.maxTake`](./#graphql)): Sets the number of items to show per page in the list view.
```
Comment (mitchellhamilton) [inline docs/pages/docs/config/lists.md:107]:
```suggestion
  - `pageSize` (default: lower of `50` or [`graphql.maxTake`](#graphql)): Sets the number of items to show per page in the list view.
```

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**